### ISSUE TITLE
per vertex normal weird output

### ISSUE THREAD

**Issue by Tomas-Ha** [dilevin/computer-graphics-meshes#47]
my per vertex normal code creates a weird output. Is this what is should look like?
<img width="656" alt="Screenshot 2023-10-20 at 10 08 28 PM" src="https://github.com/dilevin/computer-graphics-meshes/assets/90397885/6b65ce33-a826-416e-a3ed-3db70bc5c14e">
<img width="656" alt="Screenshot 2023-10-20 at 10 08 32 PM" src="https://github.com/dilevin/computer-graphics-meshes/assets/90397885/bfad1caf-d73a-47f1-bfa8-c2baa2802c1b">

**Comment by Tomas-Ha** [comment]
Specifically, the edge looks weird to me. I don't know if this is what it should look like. It has a weird rounding effect.

**Comment by Zhecheng-Wang** [comment]
It looks fine to me -- though the shading looks a bit weird. Can you double-check to make sure the normal is pointing outward?

**Comment by Tomas-Ha** [comment]
I am pretty confident that the normals are pointing outwards. When I flip the normal vectors, it removes all the lighting, and it creates this weird shading:

<img width="656" alt="Screenshot 2023-10-21 at 9 39 43 AM" src="https://github.com/dilevin/computer-graphics-meshes/assets/90397885/571c7e6b-378b-46a9-b9de-6b623e78f5a6">
<img width="656" alt="Screenshot 2023-10-21 at 9 39 48 AM" src="https://github.com/dilevin/computer-graphics-meshes/assets/90397885/f574abd9-7013-4326-a6f0-e244bdeb9edd">

**Comment by Zhecheng-Wang** [comment]
Note that triangle normal should return a normal of length $A$ where $A$ is the triangle area. Did you divide the cross product by 2?

**Comment by Tomas-Ha** [comment]
Yes I did, but for the vertex and corner, it is just a normal vector right?

**Comment by Zhecheng-Wang** [comment]
Yes. If you are worried about how the shading looks segmented on top, it is expected for per-vertex normal.
I think all looks good.
